nelox/pretty_printer.py

```diff
@@ -1,12 +1,27 @@
 from nelox.Expr import Literal, Variable, List
 
-def pretty(expr):
+def pretty(expr, ind = 0):
     if isinstance(expr, Literal):
         return str(expr.value)
     if isinstance(expr, Variable):
         return expr.name.lexeme
     if isinstance(expr, List):
-        return f"({' '.join(pretty(e) for e in expr.elements)})"
+        if func_recognizer(expr):
+            header = f"(func {pretty(expr.elements[1])} {pretty(expr.elements[2])}"
+            body_lines = [
+                "   " * (ind + 1) + pretty(e, ind + 1) for e in expr.elements[3:]
+            ]
+            if body_lines:
+                body_lines[-1] += ")"
+                return f"{header}\n" + "\n".join(body_lines)
+            else:
+                return f"{header})"
+        return f"({' '.join(pretty(e, ind) for e in expr.elements)})"
 
 def pretty_program(statements: list) -> str:
-    return "\n".join(pretty(stmt) for stmt in statements)
\ No newline at end of file
+    return "\n".join(pretty(stmt) for stmt in statements)
+
+def func_recognizer(expr):
```

Comment:
`if .. return True` is an antipattern, and here it is also unfortunate that the other case returns `None` instead of `False`.  I would recommend either

```
return expr.elements and ....
```

or

```
if not expr.elements:
    return False
...
return True
```